exchange/auction_test.go

```diff
@@ -209,15 +209,10 @@ func runCacheSpec(t *testing.T, fileDisplayName string, specData *cacheSpec) {
 			for j := 0; j < len(cache.items); j++ {
 				if formattedExpectedData == string(cache.items[j].Data) &&
 					specData.ExpectedCacheables[i].TTLSeconds == cache.items[j].TTLSeconds &&
-					specData.ExpectedCacheables[i].Type == cache.items[j].Type {
-					if specData.ExpectedCacheables[i].Type == prebid_cache_client.TypeJSON {
-						matched++
-					}
-					if specData.ExpectedCacheables[i].Type == prebid_cache_client.TypeXML &&
-						len(specData.ExpectedCacheables[i].Key) <= len(cache.items[j].Key) &&
-						specData.ExpectedCacheables[i].Key == cache.items[j].Key[:len(specData.ExpectedCacheables[i].Key)] {
-						matched++
-					}
+					specData.ExpectedCacheables[i].Type == cache.items[j].Type &&
+					len(specData.ExpectedCacheables[i].Key) <= len(cache.items[j].Key) &&
+					specData.ExpectedCacheables[i].Key == cache.items[j].Key[:len(specData.ExpectedCacheables[i].Key)] {
+					matched++
```

Comment:
This is a very minor complaint and probably not worth changing.  I just realized that the key comparison could give a false positive if the expected key (prefix) is a subset of the actual key (prefix).  E.g., if expected key is 13_sports_22 but the actual key prefix is 13_sports_222, the test would pass incorrectly.  So, ideally, an underscore should be the last character of the expected key and/or (preferably) the logic here should add an underscore to the end of the expected key if one is not already present before comparing to the actual key.  In practice, I'm doubt it's worth the trouble.  The current logic already achieves a greater precision than what we had before.

Reply:
I see. It's a fair point I can still change it if you want.

Reply:
It's hard to imagine a PBS bug sneaking through because of this, though it's theoretically possible.  If a test had two keys where one was a subset of the other (like 13_sports_22 and 13_sports_222), it would fail because the cached key matching the latter would also match the former and so throw off the count.  However, that failure would be found by the person introducing the test.  IMHO, I don't think it's worth the trouble but we'll need at least one more reviewer to agree.  :-)